testkit/src/lib.rs

```diff
@@ -596,31 +581,36 @@ impl TestKit {
         I: IntoIterator<Item = Box<Transaction>>,
     {
         let tx_hashes: Vec<_> = {
-            let mut mempool = self.mempool.write().expect(
-                "Cannot write transactions to mempool",
-            );
-
-            let snapshot = self.snapshot();
-            let schema = CoreSchema::new(&snapshot);
-            txs.into_iter()
-                .filter(|tx| tx.verify())
-                .map(|tx| {
-                    let tx_id = tx.hash();
-                    assert!(
-                        !schema.transactions().contains(&tx_id),
-                        "Transaction is already committed: {:?}",
-                        tx
-                    );
-                    mempool.insert(tx_id, tx);
-                    tx_id
-                })
-                .collect()
+            let blockchain = self.blockchain_mut();
+            let mut fork = blockchain.fork();
+            let hashes = {
+                let mut schema = CoreSchema::new(&mut fork);
+
+                txs.into_iter()
+                    .filter(|tx| tx.verify())
+                    .map(|tx| {
+                        let tx_id = tx.hash();
+                        assert!(
+                            !schema.transactions().contains(&tx_id) ||
+                                schema.transactions_pool().contains(&tx_id),
```

Comment:
I think logical not is missing here.... It's hard to understand this piece of code, let's use temporal var?